### PR TITLE
Intial Demo PR: Added OllamaInstance and relevant trait functions

### PR DESCRIPTION
**Pull request by destrex271**
So here I have used the same approach that I utilized in [LLMQueue](https://github.com/destrex271/llmqueue), adding a starting point to interact with Ollama. 
This demo implementation will use the [ollama_rs](https://github.com/pepperoni21/ollama-rs) crate.
This PR is to just start some work on the issue #85

### CODE REVIEW

File: core/src/worker/base.rs [hunk 1]
```diff
@@ -96,6 +96,13 @@ async fn execute_job(
             &msg.message.inputs,
             cfg.openai_api_key.clone(),
         )?,
+        // Adding default model URL for now
+        // Need to decide how the model url will be defined
+        "llama2" => ollama::prepare_ollama_embedding_request(
+            job_meta.clone(), 
+            &msg.message.inputs, 
+            "https://0.0.0.0:11434".to_string()
```
Comment (ChuckHend) [inline core/src/worker/base.rs:104]: Can we read this from a GUC? All these settings are defined in [guc.rs](https://github.com/tembo-io/pg_vectorize/blob/0fa65e92cc7eedee660fdddbf5a6efe62ea780e7/extension/src/guc.rs#L17)

Comment (destrex271) [reply core/src/worker/base.rs:104]: Sure!

File: extension/src/guc.rs [hunk 2]
```diff
@@ -41,6 +43,26 @@ pub fn init_guc() {
...
+        "vectorize.ollama_port",
+        "Port number",
+        "Port number on host where ollama is running. Default value is 11434",
+        &OLLAMA_PORT,
+        0,
+        65535,
+        GucContext::Suset,
+        GucFlags::default(),
+    );
+
```
[the remark this answers is not in the document]
Comment (destrex271) [reply extension/src/guc.rs:65]: Sounds good. We'll need to separate out the port number and host from this url when we create an Ollama Instance to connect to the llm instance.
Will make the changes.

File: core/src/types.rs [hunk 3]
```diff
@@ -243,6 +246,7 @@ impl FromStr for ModelSource {
         match s.to_lowercase().as_str() {
             "openai" => Ok(ModelSource::OpenAI),
             "sentence-transformers" => Ok(ModelSource::SentenceTransformers),
+            "llama2" => Ok(ModelSource::Ollama),
```
Comment (ChuckHend) [inline core/src/types.rs:249]: Would this be "ollama", then a use can request to use llama2 by providing `chat_model => 'ollama/lamma2'`?

**Comment by ChuckHend** [comment]
We branched https://github.com/tembo-io/pg_vectorize/pull/103 off this (#88) PR branch and reduced scope to just chat completion. Closing this PR in favor of https://github.com/tembo-io/pg_vectorize/pull/103

**Pull request closed without merging**